recipe/2024-06-private-bicep-registries.md

```diff
@@ -0,0 +1,425 @@
+  type: RegistryAuthType
+}
+```
+Pros:
+- Works well with current recipe config secrets design.
+
+Cons:
+- Adding a new property to specify the `type` of container registry authentication.
+- Incorrect configurations (e.g: clientId is not provided as a secret key for type azureWorkloadIdentity) are not known until recipe deployment.
+
+Option 2:
+
+Adding support to the `Applications.Core/secretStores` resource to include additional types. Today `secretStores` resource only supports types `certificate` and `generic`.
+
+secretstores.tsp
+
+```diff
+@doc("The type of SecretStore data")
+enum SecretStoreDataType {
+  @doc("Generic secret data type")
+  generic,
+
+  @doc("Certificate secret data type")
+  certificate,
+
++  @doc("basicAuthentication type is used to represent username and password based authentication and the secretstore resource is expected to have the keys 'username' and 'password'")
++  basicAuthentication
++
++  @doc("azureWorkloadIdentity type is used to represent registry authentication using azure federated identity and the secretstore resource is expected to have the keys 'clientId' and 'tenantId'")
++  azureWorkloadIdentity
++
++  @doc("awsIRSA type is used to represent registry authentication using AWS IRSA.")
++  awsIRSA
+}
+```
+Pros:
+- Leveraging the existing property `type` in `Application.core/secretStores` to specify the kind of authentication.
+- Incorrect configurations(e.g: clientId is not provided as a secret key for type azureWorkloadIdentity) can be detected during the creation of the `Applications.Core/secretStore`.
+
+Cons:
+- This options lead to either having driver specific code in engine/secretLoader or driver an making api call to corerp to get secrets information.
+
+I propose adding `type` property to the `RegistrySecretConfig` i.e Option 1, as with the current implementation of the secrets flow between engine-driver,  Option 2 lead to either having driver specific code in engine/secretLoader or driver an making api call to corerp to get secrets information. 
+
+***Bicep Example***
+
+environment.bicep
+```diff
+recipeConfig: {
+  terraform: {
+    authentication: {
+      git: {
+        pat: {
+          'github.com':{
+            secret: moduleSecrets.id
+          }
+        }
+      }
+    }
+  },
+
++  bicep: {
++    authentication:{
++      test.azurecr.io:{
++        "secretSource": "/planes/radius/local/resourcegroups/default/providers/Applications.Core/secretStores/acr-secret"
++        "type": "azureFederatedIdentity"
++      },
++      123456789012.dkr.ecr.us-west-2.amazonaws.com:{
++        "secretSource": "/planes/radius/local/resourcegroups/default/providers/Applications.Core/secretStores/aws-secret",
++        "type":"awsIRSA"
++      },
++      ghcr.io:{
++        "secretSource": "/planes/radius/local/resourcegroups/default/providers/Applications.Core/secretStores/ghcr-secret",
++        "type": "basicAuthentication"
++      },
++    }
++  }
```

Comment:
will this become a breaking change for users who have already set up [private Terraform repos](https://docs.radapp.io/guides/recipes/terraform/howto-private-registry/)?